src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -1434,14 +1434,22 @@ void TurboAssembler::li(Register rd, Operand j, LiFlags mode) {
   DCHECK(!j.is_reg());
   BlockTrampolinePoolScope block_trampoline_pool(this);
   if (!MustUseReg(j.rmode()) && mode == OPTIMIZE_SIZE) {
-    if (!FLAG_disable_constant_pool && li_count(j.immediate()) >= 4) {
+    UseScratchRegisterScope temps(this);
+    int count = li_estimate(j.immediate(), temps.hasAvailable());
+    int reverse_count = li_estimate(~j.immediate(), temps.hasAvailable());
+    if (!FLAG_disable_constant_pool && count >= 4 && (count - reverse_count) <= 1) {
```

Comment:
Can you explain why `(count - reverse_count) <= 1` is included in this check?

Reply:
Oh, I see. In case `count >= 4` but `reverse_count <= 4`. I think it would be more clear if that was more explicit in the check:
```
    if (!FLAG_disable_constant_pool && count >= 4 && reverse_count >= 4) {
```

Reply:
It should be clearer. I'll revise it immediately